runtime/lua/vim/ui.lua

```diff
@@ -103,4 +103,51 @@ function M.input(opts, on_confirm)
   end
 end
 
+
+--- Prompts the user to pick an option from a list of choices
+---
+---@param msg string Text asked
+---@param opts table
+---     - choices (table|nil)
+---		  List of options, the first letter is mapped as shortcut automatically
+---		  but it can be set by placing `&` before the chosen letter.
+---		  See |confirm()| for additional informations.
+---     - default (number|nil)
+---               Default option selected when pressing `Enter` without pressing any shortcut.
+---@param on_validate function ((idx|nil choice|nil) -> ())
+---               Called once the user chooses or abort the input.
+---               `idx` is the 1-based index of the 'choice' in `opts.choices` table,
+---               (it might be `0` if none of the choices is picked, or
+---               `nil` if the user aborted the dialog.)
+---		  `choice` is what the user choosed (`nil` on abortion or
+---		  if user did not choose any option)
```

Comment:
This looks mostly good, only 2 comments:

1. Avoid the term abort, and *especially* the term "abortion". Use "cancel" and "canceled" instead.
2. "choosed" should be "chose"

Reply:
My english is not perfect, I'll take that in mind, thanks for the advice